src/javascript/binary/pages/trade/message.js

```diff
@@ -3,33 +3,97 @@
  * events based on type of response
  */
 var Message = (function () {
-    'use strict';
+    "use strict";
 
     var process = function (msg) {
-        var response = JSON.parse(msg.data);
-        if (response) {
-            var type = response.msg_type;
-            if (type === 'authorize') {
-                TUser.set(response.authorize);
-                TradeSocket.send({ payout_currencies: 1 });
-            } else if (type === 'active_symbols') {
+
+        var response;
+
+        try {
+            response  = JSON.parse(msg.data);
+            if("object" !== typeof response || !("msg_type" in response)) {
+                throw new Error("Response from WS API is not well formatted.");
+            }
+        } catch(e) {
+            throw new Error("Response from WS API is not well formatted.");
+        }
+
+        var msg_type = response.msg_type;
+        
+        switch(msg_type) {
+
+            case "error":
+
+                // change_password
+                if("change_password" in response.echo_req) {
+                    // e.g.: {"echo_req":{"old_password":"1111","new_password":"zzzzzzzz","change_password":"1"},"version":3,"error":{"code":"InputValidationFailed","message":"Input validation failed  - $.old_password: does not match the regex pattern ^.{5,25}$"},"msg_type":"error"}
+                    PasswordWS.apiResponse(response);
+                    break;
+                }
```

Comment:
Some error responses are stateless. When we receive a response and trigger the `process` function we do not know what part of the code the request originated from (what API method was called). For non-error responses we look at the `msg_type` sub-key to dispatch the data to the right handler. However, in case of an error, `msg_type` itself becomes `error` and our only clue is lost. We cannot assume that for all API methods we could use the same error handler. Each method should be able to decide how to react to errors.

This could be solved with two approaches.
    - Switching to live-api, where the method calls are `thenable`
    - Having API return the method name when an error occurs
For now though, we have to do it as done here.